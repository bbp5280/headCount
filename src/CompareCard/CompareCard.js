import React from 'react';
import styles from '../index.css';
import PropTypes from 'prop-types';

const CompareCard = ({ comparisonData, getAvg }) => {
  return (
    <div className="compare-card">
      <section className="district1">
        <h4 className="district-name">{getAvg.district1}</h4>
        <h5 className="district-avg">Average: {getAvg.district1Average}</h5>
      </section>
      <h3>{comparisonData.compared}</h3>
      <section className="district2">
        <h4 className="district-name">{getAvg.district2}</h4>
        <h5 className="district-avg">Average: {getAvg.district2Average}</h5>
      </section>
    </div>
  );
};

CompareCard.propTypes = {
  comparisonData: PropTypes.object.isRequired
};

export default CompareCard;
