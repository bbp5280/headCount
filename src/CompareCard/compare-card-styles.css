.compare-card {
  width: 75%;
  display: block;
  margin: 0 auto;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  border: 1px solid #e4e4e4;
  border-radius: 5px;
  background-color: #fff;
  padding: 12px;
  margin-top: 20px; }

.district-name {
  color: #e64f40;
  font-size: 115%;
  text-align: center;
  margin: 0 12px 3px 12px; }

.district-avg {
  text-align: center; }

@media (max-width: 950px) {
  .compare-card {
    width: 85%; } }

@media (max-width: 650px) {
  .compare-card {
    width: 95%; } }
