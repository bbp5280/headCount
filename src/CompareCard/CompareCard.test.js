import React from 'react';
import ReactDOM from 'react-dom';
import CompareCard from './CompareCard';
import { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import config from '../../test/setup.js';

describe('COMPARE CARD', () => {
  let wrapper = shallow(<CompareCard />);

  it('should have a container', () => {
    let compareDiv = wrapper.find('.compare-card');

    expect(compareDiv.length).toEqual(1);
  });

});
