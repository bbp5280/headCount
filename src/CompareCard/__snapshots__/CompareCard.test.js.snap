exports[`COMPARE CARD should match the CompareCard snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareCard
    comparisonData={
        Object {
            "ACADEMY 20": 0.407,
            "COLORADO": 0.53,
            "compared": 0.768,
          }
    }
    getAvg={
        Object {
            "district1": "ACADEMY 20",
            "district1Average": 0.407,
            "district2": "COLORADO",
            "district2Average": 0.768,
          }
    } />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="district1">
          <h4
                    className="district-name">
                    ACADEMY 20
          </h4>
          <h5
                    className="district-avg">
                    Average: 
                    0.407
          </h5>
</section>,
        <h3>
          0.768
</h3>,
        <section
          className="district2">
          <h4
                    className="district-name">
                    COLORADO
          </h4>
          <h5
                    className="district-avg">
                    Average: 
                    0.768
          </h5>
</section>,
      ],
      "className": "compare-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              className="district-name">
              ACADEMY 20
</h4>,
            <h5
              className="district-avg">
              Average: 
              0.407
</h5>,
          ],
          "className": "district1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "ACADEMY 20",
              "className": "district-name",
            },
            "ref": null,
            "rendered": "ACADEMY 20",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Average: ",
                0.407,
              ],
              "className": "district-avg",
            },
            "ref": null,
            "rendered": Array [
              "Average: ",
              0.407,
            ],
            "type": "h5",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": 0.768,
        },
        "ref": null,
        "rendered": 0.768,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              className="district-name">
              COLORADO
</h4>,
            <h5
              className="district-avg">
              Average: 
              0.768
</h5>,
          ],
          "className": "district2",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "COLORADO",
              "className": "district-name",
            },
            "ref": null,
            "rendered": "COLORADO",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Average: ",
                0.768,
              ],
              "className": "district-avg",
            },
            "ref": null,
            "rendered": Array [
              "Average: ",
              0.768,
            ],
            "type": "h5",
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="district1">
            <h4
                        className="district-name">
                        ACADEMY 20
            </h4>
            <h5
                        className="district-avg">
                        Average: 
                        0.407
            </h5>
</section>,
          <h3>
            0.768
</h3>,
          <section
            className="district2">
            <h4
                        className="district-name">
                        COLORADO
            </h4>
            <h5
                        className="district-avg">
                        Average: 
                        0.768
            </h5>
</section>,
        ],
        "className": "compare-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                className="district-name">
                ACADEMY 20
</h4>,
              <h5
                className="district-avg">
                Average: 
                0.407
</h5>,
            ],
            "className": "district1",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "ACADEMY 20",
                "className": "district-name",
              },
              "ref": null,
              "rendered": "ACADEMY 20",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Average: ",
                  0.407,
                ],
                "className": "district-avg",
              },
              "ref": null,
              "rendered": Array [
                "Average: ",
                0.407,
              ],
              "type": "h5",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": 0.768,
          },
          "ref": null,
          "rendered": 0.768,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                className="district-name">
                COLORADO
</h4>,
              <h5
                className="district-avg">
                Average: 
                0.768
</h5>,
            ],
            "className": "district2",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "COLORADO",
                "className": "district-name",
              },
              "ref": null,
              "rendered": "COLORADO",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Average: ",
                  0.768,
                ],
                "className": "district-avg",
              },
              "ref": null,
              "rendered": Array [
                "Average: ",
                0.768,
              ],
              "type": "h5",
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
